src/fwrite.c

```diff
@@ -817,8 +829,15 @@ void fwriteMain(fwriteMainArgs args)
       }
       // compress buffer if gzip
       if (args.is_gzip && !failed) {
-        myzbuffUsed = myzbuffSize;
-        failed = compressbuff(myzBuff, &myzbuffUsed, myBuff, (int)(ch-myBuff));
+        z_stream mystream;
+        if(init_stream(&mystream)) {
```

Comment:
@philippechataignon Can this `init_stream` be raised up? It's being redone by each thread for every chunk currently, iiuc.

Reply:
Actually the logic is the same than in the the previous version with no overhead : for each chunk, a zstream is initialized, deflate is called with a Z_FINISH flush option, and deflateEnd is called. Each chunk is compressed independently (good for speed but not for the compression ratio) and written in order. With Z_FINISH, zlib manual says " After deflate has returned Z_STREAM_END, the only possible operations on the stream are deflateReset or deflateEnd.". I don't try deflateReset : perhaps it's possible to use it and to have only one allocation by thread, I will have a look. I'm quite sure that each thread must have its own ztream structure : it can't be before parallel region.